File: mapscript/csharp/examples/drawmap.cs
using System;
	
/**
 * <p>Title: Mapscript shape dump example.</p>
 * <p>Description: A C# based mapscript example to create an image given a mapfile.</p>
 * @author Yew K Choo 	ykchoo@geozervice.com
 * @version 1.0  
*/

/// <summary>
/// A C# based mapscript mapscript example to create an image given a mapfile.
/// </summary>
class DrawMap
{
  public static void usage() 
  { 
	Console.WriteLine("usage: DrawMap {mapfile} {outfile}");
	System.Environment.Exit(-1);
  }
		  
  public static void Main(string[] args)
  {
    if (args.Length != 2) usage();
    
	mapObj m_obj = new mapObj(args[0]);	
	Console.WriteLine ("# Map layers " + m_obj.numlayers + "; Map name = " + m_obj.name);	
	for (int i=0; i<m_obj.numlayers; i++) 
	{
	  Console.WriteLine("Layer [" + i + "] name: " + m_obj.getLayer(i).name);
	}
	
    imageObj i_obj = m_obj.draw();
	Console.WriteLine("Image URL = " + i_obj.imageurl + "; Image path = " + i_obj.imagepath);    
	Console.WriteLine("Image height = " + i_obj.height + "; width = " + i_obj.width); 
	try 
	{
	  i_obj.save(args[1],m_obj);
    } 
	catch (Exception ex) 
	{
                Console.WriteLine( "\nMessage ---\n{0}", ex.Message );
                Console.WriteLine( 
                    "\nHelpLink ---\n{0}", ex.HelpLink );
                Console.WriteLine( "\nSource ---\n{0}", ex.Source );
                Console.WriteLine( 
                    "\nStackTrace ---\n{0}", ex.StackTrace );
                Console.WriteLine( 
                    "\nTargetSite ---\n{0}", ex.TargetSite );	}	
  }
}

File: src/mapscript/java/examples/DrawMap.java
import edu.umn.gis.mapscript.*;

/**
 * <p>Title: Mapscript shape dump example.</p>
 * <p>Description: A Java based mapscript example to create an image given a mapfile.</p>
 * @author Yew K Choo 	ykchoo@geozervice.com
 * @version 1.0
 */

public class DrawMap {

  public static void usage() {
    System.err.println("Usage: DrawMap {mapfile} {outfile}");
    System.exit(-1);
  }

  public static void main(String[] args) {
    if (args.length != 2) usage();
   
    mapObj map = new mapObj(args[0]);
    //map.getImagecolor().setRGB(153, 153, 204);
    //styleObj st = map.getLayer(1).getClass(0).getStyle(0);
    //st.getColor().setHex("#000000");
    if( map.getLayer(1).getMetadata().get("hidden") != null ) {
    	System.out.println("Layer 1 is hidden? "+map.getLayer(1).getMetadata().get("hidden"));
    }
    int i=0;
    //for (i=0; i<100; i++) {
    	imageObj img = map.draw();
    	System.out.println("Image size is: "+img.getSize());
    	System.out.println("Image size from getBytes is: "+img.getBytes().length);
    	System.out.println(i+") the map will be drawn to:"+args[1]);
    	img.save(args[1], map);
    //}
    img.delete();
    map.delete();
    //mapscript.msCleanup();
  }
}
